src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketPostBuildStatus.java

```diff
@@ -28,27 +29,37 @@ public BitbucketPostBuildStatus(String serverId) {
         this(serverId, new DefaultJenkinsProvider());
     }
 
-    @Override
-    public void decorateCheckoutCommand(GitSCM scm, Run<?, ?> run, GitClient git, TaskListener listener,
-                                        CheckoutCommand cmd) throws GitException {
+    public Revision decorateRevisionToBuild(GitSCM scm, Run<?, ?> run, GitClient git, TaskListener listener, Revision marked, Revision rev) throws GitException {
         Injector injector = jenkinsProvider.get().getInjector();
         if (injector == null) {
             listener.getLogger().println("Injector could not be found while creating build status");
-            return;
+            return rev;
         }
 
         BuildData buildData = scm.getBuildData(run);
+
+        String ref = rev.getBranches().stream()
+                .map(branch -> {
+                    String name = branch.getName();
+                    if (StringUtils.startsWith(name, "refs/")) {
+                        return name;
+                    }
+                    return "refs/heads/" + name;
+                })
+                .findFirst()
+                .orElse(null);
         if (buildData == null || buildData.lastBuild == null) {
-            listener.getLogger().println("Build data was not found while creating a build status");
-            return;
+            run.addAction(new BitbucketRevisionAction(ref, marked.getSha1String(), serverId));
```

Comment:
This doesn't make sense to me. We get the `ref` off the `rev`, but we're posting it to the `marked`?

Reply:
Oversight, I've changed it